homeassistant/components/trello/config_flow.py

```diff
@@ -0,0 +1,220 @@
+"""Config flow for Trello integration."""
+from typing import Any
+
+from trello import Member, TrelloClient, Unauthorized
+import voluptuous as vol
+from voluptuous.schema_builder import Schema
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+
+from . import TrelloAdapter
+from .const import (
+    CONF_API_TOKEN,
+    CONF_OPTIONS_BOARDS,
+    CONF_USER_EMAIL,
+    CONF_USER_ID,
+    DOMAIN,
+    LOGGER,
+    USER_INPUT_BOARD_IDS,
+)
+
+CREDS_FORM_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_API_TOKEN): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the initial setup of a Trello integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.api_key: str = ""
+        self.api_token: str = ""
+        self.user_email: str = ""
+        self.user_id: str = ""
+        self.ids_boards: dict[str, dict[str, str]] = {}
+        self.trello_adapter: TrelloAdapter
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Create the options flow."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Prompt user for Trello API credentials.
+
+        :param user_input: None initially or users api_key and api_token.
+        :return:
+        """
+        if user_input is None:
+            return await self._show_creds_form()
+
+        self.api_key = user_input[CONF_API_KEY]
+        self.api_token = user_input[CONF_API_TOKEN]
+        self.trello_adapter = _create_trello_adapter(
+            user_input[CONF_API_KEY], user_input[CONF_API_TOKEN]
+        )
+
+        try:
+            member = await self._get_current_member()
+        except Unauthorized as ex:
+            return await self._show_error_creds_form(ex)
+
+        self.user_id = member.id
+        self.user_email = member.email
+        self.ids_boards = await self._fetch_all_boards()
+
+        return await self._show_board_form()
```

Comment:
Not sure if GitHub orders all my comments like they are added, but to get rid of these small functions you can invert this method. Only if user input is filled go into the if section. https://github.com/home-assistant/core/blob/dev/homeassistant/components/lastfm/config_flow.py is an example.